MaterialLib/SolidModels/Ehlers-impl.h

```diff
@@ -740,6 +740,94 @@ SolidEhlers<DisplacementDim>::integrateStress(
         tangentStiffness)};
 }
 
+template <int DisplacementDim>
+std::vector<typename MechanicsBase<DisplacementDim>::InternalVariable>
+SolidEhlers<DisplacementDim>::getInternalVariables() const
+{
+    return {
+        {"damage.kappa_d", 1,
+         [](typename MechanicsBase<
+                DisplacementDim>::MaterialStateVariables const& state,
+            std::vector<double>& cache) -> std::vector<double> const& {
+             assert(dynamic_cast<StateVariables<DisplacementDim> const*>(
+                        &state) != nullptr);
+             auto const& ehlers_state =
+                 static_cast<StateVariables<DisplacementDim> const&>(state);
+
+             cache.resize(1);
+             cache.front() = ehlers_state.damage.kappa_d();
+             return cache;
+         }},
+        {"damage.value", 1,
+         [](typename MechanicsBase<
+                DisplacementDim>::MaterialStateVariables const& state,
+            std::vector<double>& cache) -> std::vector<double> const& {
+             assert(dynamic_cast<StateVariables<DisplacementDim> const*>(
+                        &state) != nullptr);
+             auto const& ehlers_state =
+                 static_cast<StateVariables<DisplacementDim> const&>(state);
+
+             cache.resize(1);
+             cache.front() = ehlers_state.damage.value();
+             return cache;
+         }},
+        {"eps_p.D", KelvinVector::RowsAtCompileTime,
+         [](typename MechanicsBase<
+                DisplacementDim>::MaterialStateVariables const& state,
+            std::vector<double>& cache) -> std::vector<double> const& {
+             assert(dynamic_cast<StateVariables<DisplacementDim> const*>(
+                        &state) != nullptr);
+             auto const& ehlers_state =
+                 static_cast<StateVariables<DisplacementDim> const&>(state);
+             auto const& D = ehlers_state.eps_p.D;
+
+             cache.resize(KelvinVector::RowsAtCompileTime);
+
+             // TODO make a general implementation for converting KelvinVectors
+             // back to symmetric rank-2 tensors.
```

Comment:
The difference is that here I am converting, e.g. a 6-component Kelvin vector to a 6-component vector interpreted a symm. tensor by Paraview, and not converting to a 3x3 matrix.